db/helpers/botDataUtils.js

```diff
@@ -39,6 +39,28 @@ function isJson(item) {
   return false;
 } // isJson
 
+/**
+ * Bot Data values are stored as strings but can be converted to
+ * JSON. If a JSON string is being overridden by another JSON
+ * string we merge the values rather than replace.
+ *
+ * @param {String} originalValue - Original bot data value
+ * @param {String} newValue - New bot data value
+ * @returns {String} - Bot data where values are properly merged
+ */
+function combineValue(originalValue, newValue) {
+  let output = newValue;
+
+  // If values are json, then merge values
+  if ((isJson(originalValue)) && (isJson(newValue))) {
```

Comment:
Need to be pretty explicit about what gets overwritten, e.g. if original is { "a": [1, 2, 3] } and I send { "a": [2, 3, 4] } then, the way this is written, the new value is [2, 3, 4], i.e. it isn't *merging* arrays, it's overwriting them.